tests/TestCase/ORM/QueryTest.php

```diff
@@ -1841,6 +1841,94 @@ public function testInsert()
         $this->assertEquals(2, $result->rowCount());
     }
 
+    /**
+     * Test insertAll method inserts more than one record
+     *
+     * @return void
+     */
+    public function testInsertAll()
+    {
+        $table = $this->getTableLocator()->get('articles');
+
+        $result = $table->query()
+            ->insertAll(
+                ['title'],
+                [
+                    ['title' => 'same ol same'],
+                    ['title' => 'other same ol'],
+                    ['title' => 'more of the same'],
+                ]
+            )
+            ->execute();
+
+        $result->closeCursor();
+
+        $this->assertInstanceOf('Cake\Database\StatementInterface', $result);
+        $this->assertEquals(3, $result->rowCount());
+    }
+
+    /**
+     * Test insert method with no column name strings in the values array
+     *
+     * @return void
+     */
+    public function testInsertAllWithNoColumnNames()
+    {
+        $table = $this->getTableLocator()->get('articles');
+        $result = $table->query()
+            ->insertAll(
+                ['title'],
+                [
+                    ['same ol same'],
+                    ['other same ol'],
+                    ['more of the same'],
+                ]
```

Comment:
I think we should choose one API or the other. Supporting both numerically indexed and keyed arrays seems un-necessary. Supporting only keyed arrays would let us drop the `$columns` parameter which is nice.

Reply:
I missed that note. Okay. I'll re-work it so no numerical index support. With the signature ``` public function insertAll(array $inserts, array $types = [])``` I will derive the columns from the first element/insert of the array.

Let me know if types are to be passed at all.